### PR TITLE
INFRA-2125: Cookbook refactor

### PR DESCRIPTION
**Pull request by dud225**
- Update pgbouncer.ini template
- Add systemd service unit
- Add custom resource for managing database & user configuration
- Drop support of SmartOS

### CODE REVIEW

File: recipes/configure.rb [hunk 1]
```diff
@@ -1,89 +1,48 @@
...
-
-      action [:create, :enable]
-    end
-end
-
-
 directory '/etc/pgbouncer' do
   owner node['pgbouncer']['os_user']
   mode '0755'
-  not_if { File.directory?('/etc/pgbouncer') }
```
Comment (lgillot-lamure-ledger) [inline recipes/configure.rb:47]: Why this change ?

File: recipes/configure.rb [hunk 2]
```diff
@@ -1,89 +1,48 @@
...
-  mode '644'
-  variables node['pgbouncer']
-  notifies :restart, resources(service: 'pgbouncer'), :delayed
+  mode '0644'
+  action :create_if_missing
 end
 
-template '/etc/pgbouncer/userlist.txt' do
-  source 'userlist.txt.erb'
+file '/etc/pgbouncer/users.ini' do
```
Comment (lgillot-lamure-ledger) [inline recipes/configure.rb:34]: So, what will the content of this file be if it’s not a template anymore ? What is the relation to the template users.ini.erb still present in the commit ?

File: metadata.rb [hunk 3]
```diff
@@ -1,137 +1,9 @@
...
-depends 'resource-control'
-depends 'smf'
-
-supports 'smartos'
 supports 'ubuntu'
-
-issues_url 'https://github.com/kigster/cookbook-pgbouncer-service/issues'
-source_url 'https://github.com/kigster/cookbook-pgbouncer-service'
-
-attribute 'pgbouncer/install_method',
```
Comment (lgillot-lamure-ledger) [inline metadata.rb:23]: Why remove all this good doc ?

File: recipes/configure.rb (hunk 1 above)
Comment (dud225) [reply recipes/configure.rb:47]: because Chef handles it natively: https://github.com/chef/chef/blob/9851800cb4bcc481a00c949da2a86e15a94652f6/lib/chef/provider/directory.rb#L125

File: recipes/configure.rb (hunk 2 above)
Comment (dud225) [reply recipes/configure.rb:34]: In order to ease how the configuration is crafted, I chose to put some parts of the configuration in separate files: https://github.com/LedgerHQ/chef-pgbouncer-service/blob/d5cfd23b958f2a421a94199117b6af87b3b24e30/templates/default/pgbouncer.ini.erb#L19  (for this specific case).

This file is still fully managed by a template https://github.com/LedgerHQ/chef-pgbouncer-service/blob/d5cfd23b958f2a421a94199117b6af87b3b24e30/resources/user.rb#L54 however it is called from a custom resource. As a result, if the recipe _configure.rb_ is included and no custom resource _pgbouncer_service_user_ is called then this file will not be created.

As it is mentioned within _pgbouncer.ini_ as a file to be included, PgBouncer will faill if it is missing. That's why this part only takes care to create an empty file if it is missing.

File: metadata.rb (hunk 3 above)
Comment (dud225) [reply metadata.rb:23]: Those are not official directives: https://docs.chef.io/config_rb_metadata/

File: recipes/configure.rb (hunk 1 above)
Comment (lgillot-lamure-ledger) [reply recipes/configure.rb:47]: Oh, actually, of course ><. :+1:

File: metadata.rb (hunk 3 above)
Comment (lgillot-lamure-ledger) [reply metadata.rb:23]: Ah, I see. Too bad actually ^^.

File: recipes/configure.rb (hunk 2 above)
Comment (lgillot-lamure-ledger) [reply recipes/configure.rb:34]: Mmmh, ok, thanks for the explanation !
